### PR TITLE
[Vivante] upload the Vivante non-gui example

### PR DESCRIPTION
**Pull request by helloahn**
this sample is working through pipeline api

Signed-off-by: HyoungJoo Ahn <hello.ahn@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+%build
+export C_INCLUDE_PATH=$C_INCLUDE_PATH:/usr/include/glib-2.0/:/usr/lib/glib-2.0/include/:/usr/include/nnstreamer/
+gcc src/main.c -o vivante-example-pipeline -lglib-2.0 -lcapi-nnstreamer
+
+%install
+mkdir -p %{buildroot}/usr/bin/
+install -m 755 vivante-example-pipeline %{buildroot}/usr/bin/vivante-example-pipeline
+
+%files
```
Comment (myungjoo) [inline Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: ```%{manifest} vivante-nnstreamer-sample-pipeline.manifest```

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/usr/bin/vivante-example-pipeline
```
Comment (myungjoo) [inline Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: ```s/\/usr\/bin\//%{_bindir}/```

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+License:	Priprietary
+Source0:	%{name}-%{version}.tar.gz
+Source1001:	%{name}.manifest
+
+BuildRequires:	glib2-devel
+BuildRequires:	glibc
+BuildRequires:  capi-base-common-devel
+BuildRequires:	pkgconfig(nnstreamer)
+BuildRequires:	vivante-nnstreamer-filter-inceptionv3
+BuildRequires:	nnstreamer-vivante
```
Comment (myungjoo) [inline Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: ```s/BuildRequires/Requires/```

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+Packager:	HyoungJoo Ahn <hello.ahn@samsung.com>
+License:	Priprietary
+Source0:	%{name}-%{version}.tar.gz
+Source1001:	%{name}.manifest
+
+BuildRequires:	glib2-devel
+BuildRequires:	glibc
+BuildRequires:  capi-base-common-devel
+BuildRequires:	pkgconfig(nnstreamer)
+BuildRequires:	vivante-nnstreamer-filter-inceptionv3
```
Comment (myungjoo) [inline Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: ```s/BuildRequires/Requires/``` (who named this package?)

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+Version:	1.0.0
+Release:	0
+Group:		Applications/Multimedia
+Packager:	HyoungJoo Ahn <hello.ahn@samsung.com>
+License:	Priprietary
+Source0:	%{name}-%{version}.tar.gz
+Source1001:	%{name}.manifest
+
+BuildRequires:	glib2-devel
+BuildRequires:	glibc
```
Comment (myungjoo) [inline Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: remove this (BuildRequires: glibc) or get devel of it.

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+BuildRequires:	pkgconfig(nnstreamer)
+BuildRequires:	vivante-nnstreamer-filter-inceptionv3
+BuildRequires:	nnstreamer-vivante
+BuildRequires:	capi-nnstreamer-devel
+
+%description
+Tizen Native C-API Sample-pipeline with Vivante
+
+%package -n %{name}-inceptionv3-sample-pipeline
+Summary:	The sample-pipeline for vivante inceptionv3 model
```
Comment (myungjoo) [inline Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: 1. ```%{name}-inceptionv3-sample-pipeline``` does not exist after this build.
2. ```%package -n %{name}-inceptionv3-sample-pipeline``` == ```%package inceptionv3-sample-pipeline```

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec (hunk 4 above)
Comment (myungjoo) [reply Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: Ah... it is https://github.sec.samsung.net/AIP/vivante-nnstreamer-filter  :(

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+%setup -q
+cp %{SOURCE1001} .
+
+%build
+export C_INCLUDE_PATH=$C_INCLUDE_PATH:/usr/include/glib-2.0/:/usr/lib/glib-2.0/include/:/usr/include/nnstreamer/
+gcc src/main.c -o vivante-example-pipeline -lglib-2.0 -lcapi-nnstreamer
+
+%install
+mkdir -p %{buildroot}/usr/bin/
+install -m 755 vivante-example-pipeline %{buildroot}/usr/bin/vivante-example-pipeline
```
Comment (kparichay) [inline Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: how about - install -m 755 vivante-example-pipeline %{buildroot}%{_bindir}/vivante-example-pipeline

Comment (helloahn) [reply Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: it looks better. I'll apply it

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+%files
+%manifest vivante-nnstreamer-sample-pipeline.manifest
+%{_bindir}/vivante-example-pipeline
```
Comment (kparichay) [inline Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: You can also consider packaging all the required files (model or input files) in this package itself.

Comment (helloahn) [reply Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: the model files included at another package(`vivante-nnstreamer-filter-inceptionv3`) and this sample is using it. For the image files, it looks better. I'll add it.

File: Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+%prep
+%setup -q
+cp %{SOURCE1001} .
+
+%build
+export C_INCLUDE_PATH=$C_INCLUDE_PATH:/usr/include/glib-2.0/:/usr/lib/glib-2.0/include/:/usr/include/nnstreamer/
+gcc src/main.c -o vivante-example-pipeline -lglib-2.0 -lcapi-nnstreamer
+
+%install
+mkdir -p %{buildroot}/usr/bin/
```
Comment (kparichay) [inline Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: `/usr/bin` -> `%{_bindir}` - here as well.

Comment (helloahn) [reply Tizen.platform/Vivante_pipeline_NonGUI_inceptionV3/packaging/vivante-nnstreamer-sample-pipeline.spec]: it's fixed :)

**Comment by leemgs** [comment]
Let's merge this PR because time is going. :)

**Pull request merged**